src/openrct2/entity/Duck.cpp

```diff
@@ -203,7 +203,7 @@ void Duck::UpdateSwim()
                 landZ = tile_element_height(destination);
                 waterZ = tile_element_water_height(destination);
 
-                if (z >= landZ && z == waterZ)
+                if (z > landZ && z == waterZ)
```

Comment:
This will also cause ducks to no longer swim in shallow water, not sure if that's desired. It would be better to keep `z >= landZ`, and if it's equal, check if the surface tile has 3 corners up.

Reply:
Right, that makes more sense yeah.